tracer/src/Datadog.Trace/TraceContext.cs

```diff
@@ -100,30 +100,15 @@ public void AddSpan(Span span)
             {
                 if (RootSpan == null)
                 {
-                    // first span added is the root span
+                    // first span added is the local root span
                     RootSpan = span;
 
+                    // if we don't have a sampling priority yet, make a sampling decision now
                     if (_samplingPriority == null)
                     {
-                        if (span.Context.Parent is SpanContext { SamplingPriority: { } samplingPriority })
```

Comment:
Sampling priority, origin, propagated tags, and W3C trace state are all set when we create the `TraceContext` in `Tracer.CreateSpanContext()`, so we don't need to try to set them again here.

Only if a sampling priority wasn't propagated from upstream do we need to take a sampling decision here.